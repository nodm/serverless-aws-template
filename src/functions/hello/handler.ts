import { StatusCodes } from 'http-status-codes';

const hello = async (): Promise<{ statusCode: StatusCodes; body: string }> =>
  Promise.resolve({
    statusCode: StatusCodes.OK,
    body: 'Hello World!',
  });

export const main = hello;
