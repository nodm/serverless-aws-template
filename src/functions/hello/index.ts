import { handlerPath } from '@libs/handler-resolver';

export default {
  handler: `${handlerPath(__dirname)}/handler.main`,
  memorySize: 128,
  timeout: 5,
  events: [
    {
      http: {
        method: 'get',
        path: 'hello',
      },
    },
  ],
};
