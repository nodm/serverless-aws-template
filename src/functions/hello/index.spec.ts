import hello from './index';

describe('Testing Hello lambda configuration', () => {
  it('should create Hello lambda configuration', () => {
    expect(hello).toMatchSnapshot();
  });
});
