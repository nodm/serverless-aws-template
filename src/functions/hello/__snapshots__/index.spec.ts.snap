// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Hello lambda configuration should create Hello lambda configuration 1`] = `
{
  "events": [
    {
      "http": {
        "method": "get",
        "path": "hello",
      },
    },
  ],
  "handler": "src/functions/hello/handler.main",
  "memorySize": 128,
  "timeout": 5,
}
`;
